cpp/src/arrow/json/parser.cc

```diff
@@ -967,40 +1048,37 @@ class Handler<UnexpectedFieldBehavior::InferType> : public HandlerBase {
   }
 };
 
-BlockParser::BlockParser(MemoryPool* pool, ParseOptions options,
-                         const std::shared_ptr<Buffer>& scalar_storage)
+BlockParser::BlockParser(MemoryPool* pool, const ParseOptions& options,
+                         const std::shared_ptr<ResizableBuffer>& scalar_storage)
     : pool_(pool), options_(options) {
   DCHECK(options_.unexpected_field_behavior == UnexpectedFieldBehavior::InferType ||
          options_.explicit_schema != nullptr);
   switch (options_.unexpected_field_behavior) {
     case UnexpectedFieldBehavior::Ignore: {
       auto handler = internal::make_unique<Handler<UnexpectedFieldBehavior::Ignore>>(
           pool_, scalar_storage);
-      // FIXME(bkietz) move this to an Initialize()
-      ARROW_IGNORE_EXPR(handler->SetSchema(*options_.explicit_schema));
+      ARROW_IGNORE_EXPR(handler->Init(options_.explicit_schema));
```

Comment:
Errors should definitely not be ignored. Switching to a static factory function could be a solution. Please at least add a TODO for this.

Reply:
I'll switch to the static factory